Editorial Automation: Why & How to Set Up Chat-Ops for your Own Review System on GitHub

maelle opened the pull request:
TODOS

- [ ] update language based on https://github.com/openjournals/buffy/issues/67

mpadge commented on content/blog/2022-06-02-chatops-review-system-github/index.md:
````diff
@@ -0,0 +1,144 @@
+@ropensci-review-bot add @maelle to reviewers
+```
+
+Inspiring?
````
```suggestion
Inspired?
```

mpadge commented on content/blog/2022-06-02-chatops-review-system-github/index.md:
```diff
@@ -0,0 +1,144 @@
+## What is an editorial bot generator?
+
+The editorial bot generator buffy is the tool that supports the aforementioned commands.
+It is a **Ruby codebase** that one can 
```
```suggestion
It is a **Ruby codebase** that can 
```
along with suggestions in the following lines ... :smile:

mpadge commented on content/blog/2022-06-02-chatops-review-system-github/index.md:
```diff
@@ -0,0 +1,144 @@
+The editorial bot generator buffy is the tool that supports the aforementioned commands.
+It is a **Ruby codebase** that one can 
+
+* **configure** and **extend**, 
```
```suggestion
be * **configured** and **extended**, 
```

mpadge commented on content/blog/2022-06-02-chatops-review-system-github/index.md:
```diff
@@ -0,0 +1,144 @@
+It is a **Ruby codebase** that one can 
+
+* **configure** and **extend**, 
+* then **deploy** on for instance Heroku where it is on all the time, 
```
```suggestion
* then be **deployed** on for instance Heroku where it is on all the time, 
```

mpadge commented on content/blog/2022-06-02-chatops-review-system-github/index.md:
```diff
@@ -0,0 +1,144 @@
+Every time a new issue is opened, every time an issue comment is created, their contents are sent to the deployed app.
+If it corresponds to the **regular expression** of a **registered command**, something happens: a check is launched somewhere, a database is updated, information is copied back to the GitHub issue, etc.
+
+The "visible" bot is the **GitHub account used as bot face**: commands are addressed to it, and a "Personal" Access Token of its is used to post GitHub issue comments.
```
```suggestion
The "visible" bot is the [**GitHub account used as bot face**](https://github.com/ropensci-review-bot): commands are addressed to it, and a "Personal" Access Token of its is used to post GitHub issue comments.
```

mpadge commented on content/blog/2022-06-02-chatops-review-system-github/index.md:
```diff
@@ -0,0 +1,144 @@
+Note that one of the responders allows to launch a GitHub Action Workflow which can circumvent absence of Ruby knowledge on a team -- although a pure Ruby responder would be faster.
+You can also send a call to any API, so if you can build an external API, you can really implement many things.
+
+## The editorial bot generator buffy is for you if...
```
```suggestion
## The editorial bot generator is for you if...
```

xuanxu commented on content/blog/2022-06-02-chatops-review-system-github/index.md:
```diff
@@ -0,0 +1,145 @@
+
+## What is an editorial bot generator?
+
+The editorial bot generator buffy is the tool that supports the aforementioned commands.
```
```suggestion
The editorial bot generator [Buffy](https://github.com/openjournals/buffy) is the tool that supports the aforementioned commands.
```

As here it's been introduced I would upcase Buffy and link to its url

maelle commented:
Thanks @xuanxu!

karthik commented:
Some feedback on the post. 

Does this assume someone knows we do peer-review on GitHub etc? Otherwise, it just dives straight in.

> In this post, we explain how to assess whether the editorial bot generator Buffy might be the right fit for you.

Is there an assumption that there are folks out there doing systematic peer-review on GitHub manually who would benefit from this? Is this an audience of n > 1? Also, when the editorial bot generator is first mentioned, it is not defined. As someone quite familiar with all of this, I found the lack of explanation a bit jarring. “is the tool that supports the aforementioned commands.” doesn’t quite cut it.

I’ve made some initial changes but I find the post hard to read in its current form.

karthik commented:
> be * **configured** and **extended**, 

Why does something have to configure and extended before being deployed?

maelle commented:
Thanks a lot @karthik for your valuable feedback! :star2: 

> Does this assume someone knows we do peer-review on GitHub etc? Otherwise, it just dives straight in.

Good point. I have added an intro paragraph. Does this seem sufficient?

> Is there an assumption that there are folks out there doing systematic peer-review on GitHub manually who would benefit from this? Is this an audience of n > 1? 

Yes this is what is assumed. I have now repeated one of the items of "The editorial bot generator is for you if… " in the introduction (_"Maybe you run a submission process (of papers, software, conference abstracts) that is handled in GitHub issue (or pull request) threads (or could be moved there)?"_), does this help?
As to the target audience, yes it is small, however I based my draft on the notes of the comm call where there was interest from people who might mimick our system.

> Also, when the editorial bot generator is first mentioned, it is not defined. As someone quite familiar with all of this, I found the lack of explanation a bit jarring. “is the tool that supports the aforementioned commands.” doesn’t quite cut it.

I've added sentences at the end of the introduction and rephrased the beginning of that section, does this help?
I also plan to update the language based on https://github.com/openjournals/buffy/issues/67

> I’ve made some initial changes but I find the post hard to read in its current form.

Thanks again. We might need a few back-and-forths to get this merge-ready. :ping_pong: :pray: 

> Why does something have to configure and extended before being deployed?

There is little chance the initial configuration https://github.com/openjournals/buffy/blob/main/config/settings-production.yml will be relevant for a new user. I have now put "extended" between parentheses though, does that item need further amends?
